src/api/services/category.py

```diff
@@ -0,0 +1,32 @@
+from fastapi import Depends
+
+from src.api.schemas import CaregoryRequest
+from src.core.db.models import Category
+from src.core.db.repository.category import CategoryRepository
+
+
+class CategoryService:
+    """Сервис для работы с моделью Category."""
+
+    def __init__(self, category_repository: CategoryRepository = Depends()):
+        self.__category_repository = category_repository
+
+    async def actualize_categories(self, categories: list[CaregoryRequest]) -> None:
+        to_create, to_update = [], []
+        await self.__category_repository.archive_all()
+        already_have = await self.__category_repository.get_all_ids()
+        for category in categories:
+            if category.id not in already_have:
+                to_create.append(Category(**category.dict(), archive=False))
+            else:
+                category_dict = category.dict()
+                category_dict['archive'] = False
+                to_update.append(category_dict)
```

Comment:
```suggestion
                to_update.append(Category(**category.dict(), archive=False))
```

Reply:
Тут нужен не объект типа Category, а именно словарь, потому что я переписал метод update_all базового класса репозитория. В новом синтаксисе для обновления нужны словари и я подумал, что проще сразу в метод передавать список словарей, чем список объектов, которые потом преобразовывать в словарь. Вот из этой доки пример брал:
https://docs.sqlalchemy.org/en/20/orm/queryguide/dml.html#orm-queryguide-bulk-update

Reply:
Ну тогда так: 
```suggestion
                to_update.append(**category.dict(), archive=False)
```

Reply:
Так тоже не получается. Append принимает только один аргумент. Нужно сначала добавить в словарь ключ 'archive', а уже потом добавить этот словарь в список to_update. Я думал передавать сразу Pydantic-модель, но в ней тоже нет поля archive, его в любом случае нужно будет отдельно добавлять.

Reply:
Ну сам мог бы додумать) 
```suggestion
                to_update.append({**category.dict(), archive=False})
```